### ISSUE TITLE
new-hpovserverprofile not accepting network connections

### ISSUE THREAD

**Issue by mtbmark** [HewlettPackard/POSH-HPOneView#91]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
new-hpovserverprofile -connections should create a profile with the connections that are defined.  

### Actual Behavior
new-hpovserverprofile -connecions acts as if the input is null.  See PS output below

### Steps to reproduce
use the commandlet with the -connections option.

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion`):
HPE OneView Appliance Version:  
Output from `$PSVersionTable` on your Windows Host:

Version 2.0.296.0 HPOneView.200.psm1 contains the line:
[void]$serverProfile.connections.Add($c)

Version 2.0.312.0 HPOneView.200.psm1 does not contain the above line and $serverProfile.connections never gets populated leading to the Null error when the rest API is called.

Output from commands in powershell:

PS C:\PSScripts> $profileName = 'mwtest9999'
PS C:\PSScripts> $svr = Get-HPOVServer "hpbc000, bay 9"
PS C:\PSScripts> $svrhw = Get-HPOVServerHardwareType -Name "BL460c Gen8 3"
PS C:\PSScripts> $eg = Get-HPOVEnclosureGroup -Name "ESXi Enclosure Group"
PS C:\PSScripts> $con1 = Get-HPOVNetwork "VLAN-38-FF1" | New-HPOVServerProfileCo
nnection -connectionid 1 -connectiontype Ethernet
PS C:\PSScripts> $conList = $con1
PS C:\PSScripts> $task = New-HPOVServerProfile -name $profileName -assignmentTyp
e 'server' -Server $svr -ServerHardwareType $svrhw -EnclosureGroup $eg -connecti
ons $conList -ManageBoot | Wait-HPOVTaskComplete
You cannot call a method on a null-valued expression.
At C:\Windows\system32\WindowsPowerShell\v1.0\Modules\HPOneView.200\HPOneView.2
00.psm1:39104 char:7
+                         [void]$_spt.connections.Add($c)
+                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

New-HPOVServerProfile : [Send-HPOVRequest]: The requested HTTP method is not
valid/supported.  The request method POST is not supported or not applicable
for the requested resource. URI: /rest/server-profiles/available-networks?serve
rHardwareTypeUri=/rest/server-hardware-types/4E23E004-1847-4FFD-96C2-95CD45C65C
9B&enclosureGroupUri=/rest/enclosure-groups/629194f2-fda6-46d8-a86f-07b10de8573
3
At line:1 char:9
+ $task = New-HPOVServerProfile -name $profileName -assignmentType 'server'
-Serve ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~
    + CategoryInfo          : InvalidOperation: (POST:/rest/serv...6f-07b10de8
   5733:String) [New-HPOVServerProfile], InvalidOperationException
    + FullyQualifiedErrorId : HTTP_METHOD_NOT_ALLOWED,New-HPOVServerProfile

PS C:\PSScripts>

**Comment by ChrisLynchHPE** [comment]
Yep, that is an error on my part.  I'll get this fixed for the next release. In the mean time, you can pull down the PSM1 from the repo, and make the change you noticed to fix the problem.

However, the second error you should not be getting.  The 2.0.312 release was to fix that error, and looking at the code I can't repro that issue.

**Comment by mtbmark** [comment]
Thanks Chris.  If I get time I will try adding the line in myself and retest.  If you beat me to it and get a update out before then, I will be sure to test that one.

**Comment by ChrisLynchHPE** [comment]
I'm almost done with a small update to 2.00.  But will need to see if there is anything else.

**Comment by mtbmark** [comment]
Maybe you can take a look over the code around adding a fibrechannel connection during the creation of a new profile while you are in there.  I have yet to get that working with 2.0.296.0 but have not had the time to dig deeper into it.  If all looks good I will open a new issue for that once this fix is in and I spend more time looking at it.

**Comment by ChrisLynchHPE** [comment]
There isn't a different piece of code when adding a FC versus Ethernet Connection when creating a Server Profile.  You use the `New-HPOVServerProfileConnection` helper Cmdlet to create the `Connection` Object.  You then add all of the `Connection` objects into an `Array`, then use that `Array` as the `-Connections` parameter value.

Here is an example of what you would do to add both Ethernet and FC Connections:

```PowerShell
$con1 = Get-HPOVNetwork "Net-41" -ApplianceConnection MyAppliance.domain.com | New-HPOVServerProfileConnection -id 1 -type Ethernet
$con2 = Get-HPOVNetwork "Net-42" -ApplianceConnection MyAppliance.domain.com | New-HPOVServerProfileConnection -id 2 -type Ethernet
$con3 = Get-HPOVNetwork "fabric a" -ApplianceConnection MyAppliance.domain.com | New-HPOVServerProfileConnection -id 3 -type FibreChannel
$con4 = Get-HPOVNetwork "fabric b" -ApplianceConnection MyAppliance.domain.com | New-HPOVServerProfileConnection -id 4 -type FibreChannel
$conList = $con1, $con2, $con3, $con4
New-HPOVServerProfile -Connections $conList

#Or you can omit $conlist and just specify the $con variables in the -Connections parameter
#New-HPOVServerProfile -Connections $con1, $con2, $con3, $con4
```

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 2.0.339.1] (https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v2.0.339.1).
